tools/postgres/dbtool.js

```diff
@@ -0,0 +1,447 @@
+#!/usr/bin/env node
+const { spawnSync } = require("child_process");
+const path = require("path");
+const fs = require("fs");
+const { Command } = require("commander");
+
+const schemaPath = path.join(__dirname, "schema.sql");
+const evolutionsPath = path.resolve(path.join(__dirname, "..", "..", "conf", "evolutions"));
+
+const PG_CONFIG = (() => {
+  let rawUrl = process.env.POSTGRES_URL || "postgres://postgres:postgres@127.0.0.1:5432/webknossos";
+  if (rawUrl.startsWith("jdbc:")) {
+    rawUrl = rawUrl.substring(5);
+  }
+  const url = new URL(rawUrl);
+  url.username =
+    url.username !== "" && url.username != null
+      ? url.username
+      : process.env.PGUSER != null
+      ? process.env.PGUSER
+      : "postgres";
+  url.password =
+    url.password !== "" && url.password != null
+      ? url.password
+      : process.env.PGPASSWORD != null
+      ? process.env.PGPASSWORD
+      : "postgres";
+  url.port = url.port !== "" && url.port != null ? url.port : 5432;
+
+  const urlWithoutDatabase = new URL(url);
+  urlWithoutDatabase.pathname = "";
+
+  return {
+    url: url.toString(),
+    urlWithoutDatabase: urlWithoutDatabase.toString(),
+    username: url.username,
+    password: url.password,
+    hostname: url.hostname,
+    port: url.port,
+    database: url.pathname.length > 1 ? url.pathname.substring(1) : "webknossos",
```

Comment:
could you add a comment explaining the substring(1) here? What character should be dropped?